crates/bevy_ecs/src/schedule/config.rs

```diff
@@ -42,6 +44,104 @@ where
     }
 }
 
+impl<Marker, F, E> IntoSystemConfigs<(Marker, E)> for F
+where
+    F: IntoSystem<(), Result<(), E>, Marker>,
+    E: Debug + 'static,
+{
+    fn into_configs(self) -> SystemConfigs {
+        /// System adapter that panics if the wrapped system returns an error.
+        struct Unwrap<S>(S);
+
+        #[track_caller]
+        #[inline(never)]
+        fn panic_message(name: &str, err: impl Debug) -> ! {
+            panic!("System '{name}' returned an error: {err:?}");
```

Comment:
```suggestion
            panic!("System '{name}' returned an error: {err:?}. To avoid panicking when this occurs, consider defining an error handling system, and feeding this system's return value into it via .pipe. A number of helpful pre-built logging handlers such as .warn are provided. See https://github.com/bevyengine/bevy/blob/main/examples/ecs/system_piping.rs for more information.");
```

Reply:
Possibly too verbose: feel free to trim, or make an error code and link to it.

Reply:
I really like this suggestion, though I might play with the formatting to make it play nicely with multi-line error messages.